CMakeLists.txt

```
@@ -23,9 +23,27 @@ if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
   set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "Build type (default RelWithDebInfo)" FORCE)
 endif()
 
+# Enable auto detection for system libvterm.
+option(ENABLE_LIBVTERM_AUTO_DETECTION "Enable automatic detection for system libvterm." ON)
+
 # Look for the header file.
 option(USE_SYSTEM_LIBVTERM "Use system libvterm instead of the vendored version." OFF)
 
+# Try to find the libvterm in system.
+if(ENABLE_LIBVTERM_AUTO_DETECTION AND NOT USE_SYSTEM_LIBVTERM)
+  # Try to find the vterm.h header file.
+  find_path(LIBVTERM_INCLUDE_DIR
+    NAMES vterm.h
+    )
+
+  # vterm.h is found
+  if (LIBVTERM_INCLUDE_DIR)
+    set(USE_SYSTEM_LIBVTERM ON CACHE BOOL "vterm.h is found. Use system libvterm directly." FORCE)
+  else()
+    message(NOTICE "system libvterm not found.")
```

I would change this from NOTICE to STATUS, capitalize it, and say that we will download libvterm:
```suggestion
    message(STATUS "System libvterm not found. libvterm will be downloaded and compiled as part of the build process.")`.
```